source/Halibut.Tests/Support/BackwardsCompatibility/ProxyHalibutTestBinaryRunner.cs

```diff
@@ -148,25 +150,29 @@ public class RoundTripRunningOldHalibutBinary : IDisposable
             readonly CancellationTokenSource cts;
             readonly Task runningOldHalibutTask;
             readonly TmpDirectory tmpDirectory;
+            readonly CompatBinaryStayAlive compatBinaryStayAlive;
 
             public RoundTripRunningOldHalibutBinary(
                 CancellationTokenSource cts,
                 Task runningOldHalibutTask,
                 TmpDirectory tmpDirectory,
                 int? serviceListenPort,
-                int? proxyClientListenPort)
+                int? proxyClientListenPort, CompatBinaryStayAlive compatBinaryStayAlive)
             {
                 this.cts = cts;
                 this.runningOldHalibutTask = runningOldHalibutTask;
                 this.tmpDirectory = tmpDirectory;
-                this.ServiceListenPort = serviceListenPort;
-                this.ProxyClientListenPort = proxyClientListenPort;
+                ServiceListenPort = serviceListenPort;
+                ProxyClientListenPort = proxyClientListenPort;
+                this.compatBinaryStayAlive = compatBinaryStayAlive;
             }
             public int? ServiceListenPort { get; }
             public int? ProxyClientListenPort { get; }
 
             public void Dispose()
             {
+                compatBinaryStayAlive.Dispose();
+                Thread.Sleep(10000);
```

Comment:
Do we need this sleep?

It would be better to not wait. Or if we have to wait, then can we sense what we are waiting for has finished?

Reply:
I was sure I removed this :(